varnish-operator/templates/deployment.yaml

```diff
@@ -18,7 +18,7 @@ spec:
       serviceAccountName: varnish-operator
       containers:
       - name: varnish-operator
-        image: {{ .Values.container.image | quote }}
+        image: "{{ .Values.container.image }}:{{ .Values.container.tag }}"
```

Comment:
This is a breaking change for this helm chart as anyone who was passing in the image value as `ibmcom/varnish-operator:0.34.0` will now start getting `ibmcom/varnish-operator:0.34.0:034.0`.

Not sure it is worth the complexity of making it backward compatible, but you could... If they define `container.image` then use it as it is today, else use two NEW values (maybe `container.repository` and `container.tag`).

Also - just for consideration - if you are going this far you could also break it out to three values as Bitnami does here with their nginx helm chart: https://github.com/bitnami/charts/blob/cccdab709eebb87681a668846f29ba9a85fdc51d/bitnami/nginx/values.yaml#L71-L75 (you kind of have to ignore how they wound up using those values later on as they went a little complex on that side: https://github.com/bitnami/charts/blob/cccdab709eebb87681a668846f29ba9a85fdc51d/bitnami/nginx/templates/deployment.yaml#L146, https://github.com/bitnami/charts/blob/cccdab709eebb87681a668846f29ba9a85fdc51d/bitnami/nginx/templates/_helpers.tpl#L2-L7 AND https://github.com/bitnami/charts/blob/cccdab709eebb87681a668846f29ba9a85fdc51d/bitnami/common/templates/_images.tpl#L6-L21).

But the:
```
image:
  registry: ibmcom
  repository: varnish-operator
  tag: 0.34.0
```

Is pretty good and lets people decide WHERE they will pull the image from easily. So if you wanted to maintain backward compatibility, I suppose you could support:

```
image: ibmcom/vanish-operator:034.0

OR

image:
  registry: ibmcom
  repository: varnish-operator
  tag: 0.34.0
```

But I am not sure any helm chart tries to do that today...

Reply:
There is also the concept us using a SHA instead of image tag (look at https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy)

> To make sure the Pod always uses the same version of a container image, you can specify the image's digest; replace <image-name>:<tag> with <image-name>@<digest> (for example, image@sha256:45b23dee08af5e43a7fea6c4cf9c25ccf269ee113168c19722f87876677c5cb2).

Reply:
Good catch, I ammended my commit to be retrocompatible, just as the example you provided.